src/uu/slabtop/src/parse.rs

```diff
@@ -0,0 +1,376 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+// spell-checker:ignore (words) symdir somefakedir
+
+use std::{
+    cmp::Ordering,
+    fs,
+    io::{Error, ErrorKind},
+};
+
+#[derive(Debug, Default)]
+pub struct SlabInfo {
+    #[allow(unused)] // for slabinfo checking
+    pub(crate) version: String,
+    pub(crate) meta: Vec<String>,
+    pub(crate) data: Vec<(String, Vec<u64>)>,
+}
+
+impl SlabInfo {
+    // parse slabinfo from /proc/slabinfo
+    // need root permission
+    pub fn new() -> Result<SlabInfo, Error> {
+        let content = fs::read_to_string("/proc/slabinfo")?;
+
+        Self::parse(content).ok_or(ErrorKind::Unsupported.into())
+    }
+
+    pub(crate) fn parse(content: String) -> Option<SlabInfo> {
+        let mut lines: Vec<&str> = content.lines().collect();
+
+        let version = parse_version(lines.remove(0))?;
+        let meta = parse_meta(lines.remove(0));
+        let data: Vec<(String, Vec<u64>)> = lines.into_iter().filter_map(parse_data).collect();
+
+        Some(SlabInfo {
+            version,
+            meta,
+            data,
+        })
+    }
+
+    pub fn fetch(&self, name: &str, meta: &str) -> Option<u64> {
+        // fetch meta's offset
+        let offset = self.offset(meta)?;
+
+        let (_, item) = self.data.iter().find(|(key, _)| key.eq(name))?;
+
+        item.get(offset).copied()
+    }
+
+    pub fn names(&self) -> Vec<&String> {
+        self.data.iter().map(|(k, _)| k).collect()
+    }
+
+    #[allow(unused)] // for slabinfo checking
```

Comment:
Please document why this unused

Reply:
Perhaps version checking for /proc/slabinfo files is unnecessary? It hasn't changed its format in close to 20 years since linux kernel 2.6.

Reply:
Removed.